src/Functions/ztest.cpp

```diff
@@ -0,0 +1,189 @@
+        {
+            const auto & first_column = argument;
+            if (!isString(first_column.type))
+                throw Exception{"The fifth and sixth argument of function " + getName() + " should be strings, illegal type: " + first_column.type->getName(),
+                                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT};
+
+            const ColumnPtr & arg_json = first_column.column;
+            const auto * col_json_const = typeid_cast<const ColumnConst *>(arg_json.get());
+            const auto * col_json_string
+                = typeid_cast<const ColumnString *>(col_json_const ? col_json_const->getDataColumnPtr().get() : arg_json.get());
+
+            if (!col_json_string)
+                throw Exception{"Illegal column " + arg_json->getName(), ErrorCodes::ILLEGAL_COLUMN};
+
+            return col_json_string;
+        }
+
+        static void nan(MutableColumnPtr & to)
+        {
+            const Float64 nan = std::numeric_limits<Float64>::quiet_NaN();
+            Tuple tuple({nan, nan, nan, nan});
+            to->insert(tuple);
+        }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+        {
+            MutableColumnPtr to{getReturnType()->createColumn()};
+            to->reserve(input_rows_count);
+
+            const auto * col_usevar = getColumnString(arguments[5]);
+            const ColumnString::Chars & usevar_chars = col_usevar->getChars();
+            const ColumnString::Offsets & usevar_offsets = col_usevar->getOffsets();
+
+            for (size_t row_num = 0; row_num < input_rows_count; ++row_num)
+            {
+                const UInt64 successes_x = arguments[0].column->getUInt(row_num);
+                const UInt64 successes_y = arguments[1].column->getUInt(row_num);
+                const UInt64 trials_x = arguments[2].column->getUInt(row_num);
+                const UInt64 trials_y = arguments[3].column->getUInt(row_num);
+                const Float64 props_x = static_cast<Float64>(successes_x) / trials_x;
+                const Float64 props_y = static_cast<Float64>(successes_y) / trials_y;
+                const Float64 diff = props_x - props_y;
+                const UInt64 trials_total = trials_x + trials_y;
+                const Float64 confidence_level = arguments[4].column->getFloat64(row_num);
+
+                if (successes_x == 0 || successes_y == 0
+                    || successes_x > trials_x || successes_y > trials_y
+                    || trials_total == 0
+                    || !std::isfinite(confidence_level) || confidence_level < 0.0 || confidence_level > 1.0)
+                {
+                    nan(to);
+                    continue;
+                }
+
+                std::string_view usevar{reinterpret_cast<const char *>(&usevar_chars[usevar_offsets[row_num - 1]]), usevar_offsets[row_num] - usevar_offsets[row_num - 1] - 1};
```

Comment:
This is also seems to be unnecessary. You can compare strings.

Reply:
I will modify it to compare strings.